capa/features/address.py

```diff
@@ -88,29 +89,54 @@ def __eq__(self, other):
         return (self.process, self.tid) == (other.process, other.tid)
 
     def __lt__(self, other):
+        assert isinstance(other, ThreadAddress)
         return (self.process, self.tid) < (other.process, other.tid)
 
 
-class DynamicAddress(Address):
+class CallAddress(Address):
+    """addesses a call in a dynamic execution trace"""
+
+    def __init__(self, thread: ThreadAddress, id: int):
+        assert id >= 0
+        self.thread = thread
+        self.id = id
+
+    def __repr__(self):
+        return f"{self.thread}, call(id: {self.id})"
+
+    def __hash__(self):
+        return hash((self.thread, self.id))
+
+    def __eq__(self, other):
+        assert isinstance(other, CallAddress)
+        return (self.thread, self.id) == (other.thread, other.id)
+
+    def __lt__(self, other):
+        assert isinstance(other, CallAddress)
+        return (self.thread, self.id) < (other.thread, other.id)
+
+
+class DynamicReturnAddress(Address):
     """an address from a dynamic analysis trace"""
```

Comment:
i dont quite understand what this represents or how its used.

is this used to represent the virtual address of some code in memory during the runtime trace? does it correspond to an unrelocated part of the input sample?

Reply:
i may address this and the call/address comment below in a PR that i propose to you after we merge this PR. i'd like to demonstrate how i think about this and we can compare/contrast.

no need to resolve this comment now.